### PR TITLE
Added parser for IEC 1455 files

### PR DESCRIPTION
**Pull request by werthm**
Hi
I'm happy to contribute to becquerel by adding a parser for IEC 1455 files. Data from all of the commercial acquisition software used at our institute can be exported into this format. Therefore a dedicated parser allows us to analyze data from different spectroscopy systems directly in becquerel.
Hope you like it.
Cheers,
Dominik

### CODE REVIEW

**Comment by jvavrek** [comment]
Thanks for this @werthm! Do you mind fixing the `flake8` errors in `iec1455.py`? And we can probably have `pytest-black` ignore the `.iec` files if I can remember how to modify the `setup.cfg` accordingly.

**Comment by jvavrek** [comment]
> Hi, flake8 errors should be fixed now. Running all tests locally, I get errors (AttributeError: 'Spectrum' object has no attribute 'energies') in tests/parsers_test.py for all the test file extensions, not just .iec.

Interesting, your test environment must be ignoring the option to skip plot tests. But this shows that the plot tests, which are always (?) skipped can be quite out of date, so good catch. @markbandstra @jccurtis should we address this? IIRC the fix here is just `spec.energies` --> `spec.bin_centers_kev`

**Comment by jvavrek** [comment]
@werthm  can you update the top of the top-level `.pre-commit-config.yaml` to the following:
```yaml
# Exclude CNF, cnf and iec files
exclude: '(?:\.cnf|\.CNF|\.iec)$'
```
and see if that ignores the `.iec` files correctly?

**Comment by werthm** [comment]
Done!

**Comment by werthm** [comment]
Any updates on merging this with the master?

**Comment by markbandstra** [comment]
Hi @werthm we're going to have a meeting tomorrow and will take a look. Thanks again for this contribution!

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # Exclude CNF and cnf files
```
Comment (jvavrek) [inline .pre-commit-config.yaml:1]: Update this comment to include iec

File: becquerel/parsers/iec1455.py [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+            # parse records
+            if record == 1:
+                if len(tok) == 5:  # all information
+                    data["system_identification"] = tok[0]
+                    data["sub_system_identification"] = tok[1]
+                    data["ADC_number"] = int(tok[2])
+                    data["segment_number"] = int(tok[3])
+                    data["digital_offset"] = int(tok[4])
+                elif (
+                    len(tok) == 4 and line.startswith(" ") is True
```
Comment (jvavrek) [inline becquerel/parsers/iec1455.py:118]: `is True` shouldn't be necessary here

File: becquerel/parsers/iec1455.py [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+                line = re.sub(r"(e[+-][0-9]{2})", r"\1 ", line.lower())
+                tok = line.split()
+                for coeff in tok:
+                    cal_coeff.append(float(coeff))
+            elif record == 5:
+                # fix nasty formatting of Interwinner export
+                line = re.sub(r"(e[+-][0-9]{2})", r"\1 ", line.lower())
+                tok = line.split()
+                for coeff in tok:
+                    fwhm_coeff.append(float(coeff))
```
Comment (jvavrek) [inline becquerel/parsers/iec1455.py:159]: could probably reduce code duplication here, but this is fine

**Review by markbandstra — APPROVED** [review]
Good work, thanks for the contribution @werthm ! I can merge when the tests are done, unless there is anything else you want to add.

**Comment by markbandstra** [comment]
Hi @werthm, apologies for your frustration with our process here, and thank you for your honest feedback. I consulted with @jccurtis and @jvavrek in my response here. We customarily require the author of a pull request to handle any requested changes, even if they seem trivial. Sorry for the mismatch in expectations here. We’ve tried to spell this out in the [contribution guidelines](https://github.com/lbl-anp/becquerel/blob/main/CONTRIBUTING.md) and please let us know if they can be clearer.

Looking more closely into this PR, a lot of the specific test failures were around pre-commit formatting and excluding a new file extension (*.iec) from the pre-commit pipeline. We realized that it will save us all a lot of time to enable a CI hook on GitHub to automatically run pre-commit and push relevant fixes, which we tested [here](https://github.com/lbl-anp/becquerel/pull/330)). This hook will only work for branches in the main repo so outside developer forks will need to make the fixes (run `pre-commit run --all`) themselves. Once a contribution is made we could add you to the dev team so you can push branches to the main repo.

The original intention behind this project was to pool our efforts so we each didn't have to reinvent the wheel when it comes to spectroscopic analysis, and we made it open source in case anyone else out there would find it useful too. We’re happy to see that you found it useful, and have contributed in return. This is as good as it gets in open source! On the other hand this is a side project for us and we don’t have much bandwidth, so we appreciate your patience with this process.

**Comment by markbandstra** [comment]
Merging despite test failures, which are unrelated. Those issues will be tracked by #332

**Comment by werthm** [comment]
Hi @markbandstra @jccurtis @jvavrek , thanks for your reply. I think the contribution guidelines could be a bit more detailed, especially for people unfamiliar with the various developing tools you're using. Please keep in mind that people might not be able to dedicate time to get into the various tools (for whatever reason). It would be a pity to loose new contributions just because of this.

**Pull request merged**